src/v/raft/state_machine_manager.h

```diff
@@ -84,12 +106,19 @@ class state_machine_manager final {
         }
     }
 
+    ss::future<> remove_local_state();
+
 private:
     using stm_ptr = ss::shared_ptr<state_machine_base>;
+    struct named_stm {
+        named_stm(ss::sstring, stm_ptr);
```

Comment:
Does the default not work?

Reply:
i did this to use it with `std::vector::emplace_back`